turborepo-tests/helpers/test_example.sh

```diff
@@ -0,0 +1,54 @@
+#!/usr/bin/env bash
+# Start by figuring out which example we're testing and its package manager
+example_path=$1
+package_manager=$2
+
+# If you need to do some debugging, you can set these to true
+debug=false
+force=false
+
+# Use the right command for each package manager
+if [ "$package_manager" == "npm" ]; then
+  package_manager_command="npm install"
+  update_turbo_version="npm install turbo@latest -w"
+elif [ "$package_manager" == "pnpm" ]; then
+  package_manager_command="pnpm install"
+  update_turbo_version="pnpm install turbo@latest -w"
+elif [ "$package_manager" == "yarn" ]; then
+  package_manager_command="yarn"
+  update_turbo_version="yarn upgrade turbo@latest -W"
+fi
+
+# All examples implement these two tasks
+# and it's reasonable to assume that they will continue to do so
+turbo_command="turbo build lint --output-logs=errors-only"
+
+# Head into the example directory
+cd $example_path
+
+if [ $force == true ]; then
+  echo "Forcing execution of tasks on first run..."
+fi
+
+# Do the work!
+if [ $debug == true ]; then
+  $package_manager_command
+  $update_turbo_version
+  [[ $force == true ]] && $turbo_command --force || $turbo_command
+else
+  $package_manager_command > /dev/null 2>&1
+  $update_turbo_version > /dev/null 2>&1
+  [[ $force == true ]] && $turbo_command --force || $turbo_command > /dev/null
+fi
```

Comment:
What if there's an existing cache? Do you want to assert that there's is _not_ a cache hit here, so we can be sure tests are always running from a clean slate? Since this is now running "locally" and not in an isolated environment, I'd be concerned that gitignored things can cause tests passing or failing at random.

Reply:
Isn't hitting cache valid? Same principle as caching for anything `turbo`, right? "If you've done it before, hit the cache." I'm trying to think of why I would specifically _not want it to hit cache_. Purely on principle of hermeticism for tests, I suppose?

I'm thinking back to the goals, though. "Make sure an example works out of the box." If `turbo build lint` has been ran with those cache inputs and came back clean prior to the test run, then it stands to reason that it's valid enough to pass for the test? 

I'm thinking I don't care if it came from a fresh run or from cache? At least it didn't seem to matter when I was trying to break that assumption by hand when I first wrote up this strategy, anyway.

Reply:
Hitting cache on the first run just means that you could be replaying a run even though files have changed (e.g. if actual caching is broken, and we replay a previous cache), we'd never know, but a user would when they use this example.

You could do `rm -rf .turbo/ node_modules/ || true` before `$package_manager_command` to create a clean environment.

(Note we aren't talking about cache hit on `turbo run test --filter="turborepo-examples-test-*"` , we're talking about cache hit on `turbo build lint --output-logs=errors-only` in each example.)

Reply:
Ah, okay, I see. I'll add the assertion, too. 👍